esphome/components/pipsolar/pipsolar.cpp

```diff
@@ -0,0 +1,918 @@
+          this->warning_eeprom_failed_->publish_state(value_warning_eeprom_failed_);
+        }
+        if (this->fault_inverter_over_current_) {
+          this->fault_inverter_over_current_->publish_state(value_fault_inverter_over_current_);
+        }
+        if (this->fault_inverter_soft_failed_) {
+          this->fault_inverter_soft_failed_->publish_state(value_fault_inverter_soft_failed_);
+        }
+        if (this->fault_self_test_failed_) {
+          this->fault_self_test_failed_->publish_state(value_fault_self_test_failed_);
+        }
+        if (this->fault_op_dc_voltage_over_) {
+          this->fault_op_dc_voltage_over_->publish_state(value_fault_op_dc_voltage_over_);
+        }
+        if (this->fault_battery_open_) {
+          this->fault_battery_open_->publish_state(value_fault_battery_open_);
+        }
+        if (this->fault_current_sensor_failed_) {
+          this->fault_current_sensor_failed_->publish_state(value_fault_current_sensor_failed_);
+        }
+        if (this->fault_battery_short_) {
+          this->fault_battery_short_->publish_state(value_fault_battery_short_);
+        }
+        if (this->warning_power_limit_) {
+          this->warning_power_limit_->publish_state(value_warning_power_limit_);
+        }
+        if (this->warning_pv_voltage_high_) {
+          this->warning_pv_voltage_high_->publish_state(value_warning_pv_voltage_high_);
+        }
+        if (this->fault_mppt_overload_) {
+          this->fault_mppt_overload_->publish_state(value_fault_mppt_overload_);
+        }
+        if (this->warning_mppt_overload_) {
+          this->warning_mppt_overload_->publish_state(value_warning_mppt_overload_);
+        }
+        if (this->warning_battery_too_low_to_charge_) {
+          this->warning_battery_too_low_to_charge_->publish_state(value_warning_battery_too_low_to_charge_);
+        }
+        if (this->fault_dc_dc_over_current_) {
+          this->fault_dc_dc_over_current_->publish_state(value_fault_dc_dc_over_current_);
+        }
+        if (this->fault_code_) {
+          this->fault_code_->publish_state(value_fault_code_);
+        }
+        if (this->warnung_low_pv_energy_) {
+          this->warnung_low_pv_energy_->publish_state(value_warnung_low_pv_energy_);
+        }
+        if (this->warning_high_ac_input_during_bus_soft_start_) {
+          this->warning_high_ac_input_during_bus_soft_start_->publish_state(
+              value_warning_high_ac_input_during_bus_soft_start_);
+        }
+        if (this->warning_battery_equalization_) {
+          this->warning_battery_equalization_->publish_state(value_warning_battery_equalization_);
+        }
+        this->state_ = STATE_IDLE;
+        break;
+      case POLLING_QT:
+        this->state_ = STATE_IDLE;
+        break;
+      case POLLING_QMN:
+        this->state_ = STATE_IDLE;
+        break;
+    }
+  }
+
+  if (this->state_ == STATE_POLL_CHECKED) {
+    bool enabled = true;
+    String fc;
```

Comment:
```suggestion
    std::string fc;
```

In esphome, we use the C++ stdlib `std::string` for strings. Please update the code to use that here (see here for api reference: https://en.cppreference.com/w/cpp/string/basic_string)